### PR TITLE
fix: [CDS-83434]: Fixed bugs in infra gitx

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
### Summary

1.  Fixed following bugs :
- Non-default branch infra details cancelled api call and loaded incorrect data by passing default branch
- On opening inline infra remote infra details came up 
- Save is disabled once save api fails 

2. Also fixed https://harness.atlassian.net/browse/CDS-83296

#### Screenshots

https://github.com/harness/harness-core-ui/assets/98508399/eabfa26b-3588-427b-8046-d08486ec01f9

https://github.com/harness/harness-core-ui/assets/98508399/767f9b8e-12be-4afa-a79a-1c5feecd9138

Screenshot 2023-11-07 at 3 01 02 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by sonaligoyal-harness** [comment]
trigger messagecheck

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx [hunk 1]
```diff
@@ -165,7 +165,7 @@ export default function InfrastructureDefinition({ isEnvPage }: { isEnvPage: boo
             openUnsavedChangesDiffModal={openUnsavedChangesDiffModal}
             handleInfrastructureUpdate={handleInfrastructureUpdate}
             updatedInfra={updatedInfrastructure}
-            infrastructureFetchDetails={infrastructureFetchDetails}
+            infrastructureFetchDetails={infraStoreType === 'REMOTE' ? infrastructureFetchDetails : undefined}
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx:168]: Please test this for inline infrastructure case

Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx:168]: Yes , i did it and as we discussed last time, inline flow has not been touched and it is working fine.
Issue was that the response was not getting reset and in child component we were facing issues related to this stale value for inline.

**Review by rameshx — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by samarth-sharma31 — APPROVED** [review]

**Comment by kalyansai-harness** [comment]
@vikrant-harness @samarth-sharma31 @rameshx did we verify all the videos and test cases related to this fix?

**Pull request merged**